ci/badge/gen_badge.py

```diff
@@ -0,0 +1,137 @@
+def get_code_g_y_r(val, scale):
+    """get_code_g_y_r(val, scale) -> str"""
+    if val <= 50:
+        red = 255
+        green = val * (255 / (float(scale) / 2))
+    else:
+        green = 255 * val / scale
+        red = 255 - (val - 50) * (255 / (float(scale) / 2))
+
+    rgb = (red, green, 0)
+
+    return '#%02x%02x%02x' % rgb
+
+##
+# @brief Generate a github badge svg file representing code coverage
+# @param[in] html A concatenated string of the whole contents in index.html that is the result of LCOV
+# @param[in] path A file path to save the svg file
+def gen_coverage_badge(html, path):
+    str_coverage = 'coverage'
+
+    #parse LCOV html
+    soup = BeautifulSoup(html, 'html.parser')
+    line_hits, lines, func_hits, funcs = \
+        soup.find('table').find_all('td', {'class': 'headerCovTableEntry'})
+    line_hits = float(line_hits.text)
+    lines = float(lines.text)
+    func_hits = float(func_hits.text)
+    funcs = float(funcs.text)
+    line_coverage = line_hits / lines
+    rgb_code = get_code_g_y_r(line_coverage * 100, 100)
+
+    dwg = svgwrite.Drawing(path, size=(113.3, 20))
+    dwg.viewbox(width=1133, height=200)
+
+    vert_grad = svgwrite.gradients.LinearGradient(None, end=(0, 1), id="vert_lin_grad")
+    vert_grad.add_stop_color(offset='0%', color='#FFF', opacity=.1)
+    vert_grad.add_stop_color(offset='100%', opacity=.1)
+    dwg.defs.add(vert_grad)
+
+    mask = dwg.mask(id="grad_mask")
+    mask.add(dwg.rect((0, 0), (1133, 200), rx=30, fill="#FFF"))
+    dwg.defs.add(mask)
+
+    grp_rects = dwg.g(id='grp_rects', mask="url(#grad_mask)")
+    grp_rects.add(dwg.rect((0, 0), (603, 200), fill="#555"))
+    grp_rects.add(dwg.rect((603, 0), (703, 200), fill=rgb_code))
+    grp_rects.add(dwg.rect((0, 0), (1133, 200), fill="url(#vert_lin_grad)"))
+    dwg.add(grp_rects)
+
+    if line_coverage == 0:
+        text_margin = 25
+    elif line_coverage == 1:
+        text_margin = -25
+    else:
+        text_margin = 0
+
+    style = 'text-anchor:start;font-family:Verdana,DejaVu Sans,sans-serif;font-size:110px'
+    grp_texts = dwg.g(id='grp_texts', fill='#fff', style=style)
+    grp_texts.add(dwg.text(str_coverage, (60, 148), textLength='503', fill='#000', opacity='0.25'))
+    grp_texts.add(dwg.text(str_coverage, (50, 138), textLength='503'))
+    grp_texts.add(dwg.text(format(line_coverage, "2.1%"), (703 + text_margin, 148),\
+        textLength='330', fill='#000', opacity='0.25'))
+    grp_texts.add(dwg.text(format(line_coverage, "2.1%"), (693 + text_margin, 138),\
+        textLength='330'))
+    dwg.add(grp_texts)
+    dwg.save(path)
+
+if __name__ == '__main__':
+    # argv[1]: [badgetype] a string indicating the type of badge, 'codecoverage'
+    # argv[2]: [url/file] a path or url of LCOV html to get information for badge generation
+    # argv[3]: [file] a file path to save the generated svg file
+    if len(sys.argv) < 4:
+        exit(1)
+
+    badgetype = 'unknown'
```

Comment:
Looks good to me. Please add additional badges. :)

Reply:
@leemgs As your command.

Reply:
Not command, It's proposal. :)